crates/biome_css_formatter/src/css/properties/generic_property.rs

```diff
@@ -12,6 +12,45 @@ impl FormatNodeRule<CssGenericProperty> for FormatCssGenericProperty {
             value,
         } = node.as_fields();
 
+        let is_dash_identity = name.clone().unwrap().as_css_dashed_identifier().is_some();
+
+        let has_comma = value.iter().any(|v| v.text().eq(","));
+
+        if has_comma && !is_dash_identity {
+            write!(
+                f,
+                [
+                    name.format(),
+                    colon_token.format(),
+                    indent(&format_with(|f| {
+                        let last = value.last().unwrap();
+
+                        for (idx, v) in value.iter().enumerate() {
+                            if idx == 0 {
+                                write!(f, [soft_line_break_or_space()])?;
+                            }
+
+                            if v.text().eq(",") {
+                                write!(f, [v.format(), hard_line_break()])?;
+                            } else if let Some(next) = value.iter().nth(idx + 1) {
+                                if next.text().eq(",") || v == last {
+                                    write!(f, [v.format()])?;
+                                } else {
+                                    write!(f, [v.format(), space()])?;
+                                }
+                            } else {
+                                write!(f, [v.format()])?;
+                            }
+                        }
```

Comment:
I am not really sure I understand what's happening here, so I am waiting for your to update the PR description and explain how the algorithm works.